code/parachain/frame/composable-traits/src/staking/mod.rs

```diff
@@ -41,6 +42,38 @@ pub struct Reward<Balance> {
 	pub last_updated_timestamp: u64,
 }
 
+#[derive(RuntimeDebug, PartialEq, Eq, Clone, MaxEncodedLen, Encode, Decode, TypeInfo)]
+pub enum RewardType<Balance> {
+	Earnings(),
+	RateBased(RateBasedReward<Balance>),
+}
+
+impl<Balance> RewardType<Balance> {
+	pub fn as_ref_mut_rate_based(&mut self) -> Option<&mut RateBasedReward<Balance>> {
+		match self {
+			RewardType::Earnings() => None,
+			RewardType::RateBased(ref mut rate_based_reward) => Some(rate_based_reward),
+		}
+	}
+}
+
+impl<Balance: Zero> RewardType<Balance> {
+	pub fn from_config(reward_config: RewardConfigType<Balance>, now_seconds: u64) -> Self {
+		match reward_config {
+			RewardConfigType::Earnings() => RewardType::Earnings(),
+			RewardConfigType::RateBased(rate_based_config) =>
+				RewardType::RateBased(RateBasedReward::from_config(rate_based_config, now_seconds)),
+		}
+	}
+}
+
+#[derive(RuntimeDebug, PartialEq, Eq, Clone, MaxEncodedLen, Encode, Decode, TypeInfo)]
+pub enum RewardConfigType<Balance> {
+	// REVIEW(benluelo): Consider renaming. Perhaps `ExternallyFunded` or something similar?
```

Comment:
We can also call this `ProtocolDistribution` https://www.investopedia.com/terms/d/distribution.asp

Reply:
https://github.com/ComposableFi/composable/pull/1668/commits/cc9ca2596457b95ddbee6e9e541c914b52dbc3a8